### ISSUE TITLE
Request: Support For RISC-V and PowerISA hardware

### ISSUE THREAD

**Issue by Ratatouilledummie** [ptitSeb/box86#678]
Sorry for the incovennience but can be good if box86 and box86_64 can receive support for POWERISA and RISC-V architetures these two is RISC architeture

**Comment by ptitSeb** [comment]
I'm not sure of what you are proposing here.

But I do plan to add Risc-V support to box64 dynarec yes.

**Comment by Ratatouilledummie** [comment]
sorry is that IBM's PowerISA architecture doesn't have x86 emulation software and it would be great, a support because PowerISA is a more open ISA than RISC-V

**Comment by ptitSeb** [comment]
Ah ok. So stuff like Power9?

**Comment by Ratatouilledummie** [comment]
> Ah ok. So stuff like Power9?

yes, sorry for the delay

**Comment by ptitSeb** [comment]
Ah ok.
Well, there is some support already, but no dynarec, only interpretor. And because I don't have any hardware, I will not work on the Dynarec myself (but I accept contributions)

**Comment by geerlingguy** [comment]
@ptitSeb - I have a SiFive HiFive Premier P550, and also have a working AMD RX 480 GPU on it, I'd love to do some testing with RISC-V and box86/box64... should this just work out of the box yet?

I didn't see any mention of `riscv64` in the COMPILE.md instructions: https://github.com/ptitSeb/box86/blob/master/docs/COMPILE.md

**Comment by ptitSeb** [comment]
> [@ptitSeb](https://github.com/ptitSeb) - I have a SiFive HiFive Premier P550, and also have a working AMD RX 480 GPU on it, I'd love to do some testing with RISC-V and box86/box64... should this just work out of the box yet?
> 
> I didn't see any mention of `riscv64` in the COMPILE.md instructions: https://github.com/ptitSeb/box86/blob/master/docs/COMPILE.md

Yes, Box64 will just compile fine on Riscv. And there is dynarec support too so you will get decent speed :)
you can use `ccmake ..` for an interactive way of selecting box64 build option (install with `sudo apt install cmake-curses-gui`

**Comment by ksco** [comment]
> [@ptitSeb](https://github.com/ptitSeb) - I have a SiFive HiFive Premier P550, and also have a working AMD RX 480 GPU on it, I'd love to do some testing with RISC-V and box86/box64... should this just work out of the box yet?
> 
> I didn't see any mention of `riscv64` in the COMPILE.md instructions: https://github.com/ptitSeb/box86/blob/master/docs/COMPILE.md

There is no box86 RISC-V support, but box64 works fine on RISC-V now, we had many big Windows games like The Witcher 3, Genshin Impact running on Milk-V Pioneer already. Here is the compiling document for RISC-V: https://github.com/ptitSeb/box64/blob/main/docs/COMPILE.md#for-risc-v

I'm maintaining the box64 RISC-V backend, feel free to email me if you need any help on the box64 RISC-V experiment, I'm happy to help! :)
